[doc] Add a task behavior README (#91)
ObeoNetwork/pepper: backend/pepper-domain-services/README.md

> ## Managing time zone

Reviewer: ```suggestion
## Managing time zones
```

> Pepper only support a single time zone.

Reviewer: ```suggestion
Pepper only supports a single time zone.
```

> The server and the navigator must be on the same time zone.

Reviewer: ```suggestion
The server and the browser must use the same time zone.
```

> ## Considering constraint on a task

Reviewer: ```suggestion
## Considering task constraints
```

> A task is constrained by two of TimeConstraint(START, END or DURATION).

Reviewer: ```suggestion
A task is constrained by two `TimeConstraint` values among:

- `START`
- `END`
- `DURATION`
```

> If the constrained boundary (START or END) is itself also constrained by a dependency, we consider that the dependency constraint is stronger, that the task constraint. In that case we consider that the duration is not more constraining.

Reviewer: ```suggestion
If a constrained boundary (`START` or `END`) is also constrained by a dependency, the dependency constraint is considered stronger than the task constraint.

In that case, the duration is no longer considered constraining.
```

A boundary constrained by a dependency can not be changed directly.

> For example: A START-DURATION task has its end date constrained by a dependency. If the start date is moved, the end date is let unchanged and the duration is updated.

Reviewer: ```suggestion
For example: A START-DURATION task has its end date constrained by a dependency. If the start date is moved, the end date remains unchanged and the duration is updated accordingly.
```

> ## Interaction in Gantt

Reviewer: ```suggestion
## Gantt interactions
```

> ### The task is moved

Reviewer: ```suggestion
### Moving a task
```

> - If dependencies nothing is done TODO

Reviewer: ???

Author: done

> - Otherwise the constraining boundary(ies) of the task is (are) updated according to the calculation option.

Reviewer: ```suggestion
- Otherwise, the constraining boundary or boundaries are updated according to the task calculation option:
```

>   -- START-END Both boundaries are updated. The duration may be changed if included non-working days are changed

Reviewer: ```suggestion
  -- `START-END`: both boundaries are updated. The duration may change if the number of included non-working days changes.
```

>   -- START-DURATION and END_DURATION: The constraining boundary is updated. The constrained boundary may be updated beyond the move if included non-working days are changed because duration is kept.

Reviewer: ```suggestion
  -- `START-DURATION` and `END-DURATION`: the constraining boundary is updated. The constrained boundary may move by more than the drag delta if the number of included non-working days changes, because the duration is preserved.
```

> ### One of the task boundaries(start or end) is changed

Reviewer: ```suggestion
### Changing one task boundary
```

> 

Reviewer: When one task boundary, either start or end, is changed:

> - If the boundary is constraining, the boundary is updated, the other constraint is kept.

Reviewer: ```suggestion
- If the boundary is constraining, this boundary is updated and the other constraint is preserved.
```

>   - Dependency case: If the constrained boundary is constrained by dependency, the move boundary AND the duration or changed.

Reviewer: ```suggestion
- If the opposite boundary is constrained by a dependency, the moved boundary is updated and the duration is updated as well.
```

> - Otherwise, we consider that the user want to update the duration by the move delta.

Reviewer: ```suggestion
- Otherwise, the change is interpreted as a user intent to update the duration by the move delta.
```

>   -- [FUTURE ENHANCEMENT] We could imagine forbidding this case with a global option. In that case, it would not be possible to move the task boundary

Reviewer: ```suggestion
  -- [FUTURE ENHANCEMENT] A global option could forbid changing a non-constraining boundary directly. In that mode, moving such a task boundary would not be allowed.
```
